### ISSUE TITLE
the new Admob GDPR Update

### ISSUE THREAD

**Issue by X0t0** [ratson/cordova-plugin-admob-free#223]
Hi, is there any option to the GDPR: 
https://developers.google.com/admob/android/eu-consent#forward_consent_without_the_consent_sdk

because i have no idea what to do

**Comment by anirnet** [comment]
me too

**Comment by X0t0** [comment]
@anirnet i hope someone can help us

**Comment by ivanov84** [comment]
@X0t0 You can build your own modal with agreement and pass "npa" param when users deny agreement

**Comment by X0t0** [comment]
@ivanov84 can you give me a small tutorial how to pass npa , because i don't know and i couldn't find this npa in the documentation

**Comment by jellomaster** [comment]
You'll need to handle permissions on your side.. so make a popup with the terms and agree/deny buttons.  On agree save the setting and continue as normal.  If they disagree, save the preference and add the npa:1 as adExtras

Ex
admob.banner.config({
    id: 'bannerId',
    adExtras: {npa: 1}
});

**Comment by ratson** [comment]
@jellomaster Thank you for answering this. I am really have no idea about GDPR.

It would be great if anyone could explain what needs to done for library like this, and any further actions required to comply to GDPR, it would help many users.

**Comment by jellomaster** [comment]
I don't think you need to change anything in the plugin... The optimal solution would be for another plugin.   The plugin should use some sort of database (ex. sqlite) and on startup don't display ads and load the value... if it's blank, display a modal that describes the Personalized ads feature and provide Allow / Deny buttons.  Store the user's choice in the database for the next time the app loads.  This way when displaying ads you can have an if/else section.  If personalized ads are allowed, use the normal code.  Else, use adExtras: {npa: 1} in the config like I mentioned above.

**Comment by zawavie** [comment]
@jellomaster if you could give a small example (i mean the code) of how to do it that will be so helpfull 👍 
Thanks for your reply

**Comment by tobyherrmann** [comment]
Because I had the same problem in one of my apps, I have written a new plugin yesterday night and tody, which wrapps the google consent sdk for ionic. At the moment it supports only Android.

You can get it here: https://github.com/tobyherrmann/cordova-plugin-google-consent

**Comment by tobyherrmann** [comment]
To pass the NPA parameter with typescript you have to write a custom config, because the ionic wrapper does not support it at the moment.

See #210

**Comment by prantikv** [comment]
Hope the 
`adExtras: {npa: 1}` is added as an option to the plugin @ratson

**Comment by brunoalex** [comment]
@ratson is  adExtras: {npa: 1} an option that we can use ? i did and it looks like the same ads are displayd
